panic: extra: AURConfig: key not present

### Discussed in https://github.com/orgs/goreleaser/discussions/5715

<div type='discussions-op-text'>

<sup>Originally posted by **ldez** April 12, 2025</sup>

Feels like a regression related to https://github.com/goreleaser/goreleaser/pull/5685

```
     • arch user repositories
      • cloning 9f88c5f107c9b1a9559033da1fbe66bc639bf79ef7a9060f2ea8939a7bcac314-master dist/git/9f88c5f107c9b1a9559033da1fbe66bc639bf79ef7a9060f2ea8939a7bcac314-master
      • writing                                      path=dist/git/9f88c5f107c9b1a9559033da1fbe66bc639bf79ef7a9060f2ea8939a7bcac314-master/PKGBUILD
      • pushing                                      repository=ssh://aur@aur.archlinux.org/golangci-lint-bin.git name=9f88c5f107c9b1a9559033da1fbe66bc639bf79ef7a9060f2ea8939a7bcac314 file=PKGBUILD
      • writing                                      path=dist/git/9f88c5f107c9b1a9559033da1fbe66bc639bf79ef7a9060f2ea8939a7bcac314-master/.SRCINFO
      • pushing                                      repository=ssh://aur@aur.archlinux.org/golangci-lint-bin.git name=9f88c5f107c9b1a9559033da1fbe66bc639bf79ef7a9060f2ea8939a7bcac314 file=.SRCINFO
      • took: 15s
  • took: 4m18s
panic: extra: AURConfig: key not present
goroutine 11 [running]:
github.com/goreleaser/goreleaser/v2/internal/artifact.MustExtra[...]({{0x31cb321, 0x8}, {0xc000116ca0, 0x1f}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...)
	github.com/goreleaser/goreleaser/v2@v2.8.2/internal/artifact/artifact.go:243 +0x23c
github.com/goreleaser/goreleaser/v2/internal/pipe/aur.doPublish(0xc000844008, {0xc0010ec3c0, 0x2, 0x2})
	github.com/goreleaser/goreleaser/v2@v2.8.2/internal/pipe/aur/aur.go:383 +0xab
github.com/goreleaser/goreleaser/v2/internal/pipe/aur.Pipe.Publish({}, 0xc000844008)
	github.com/goreleaser/goreleaser/v2@v2.8.2/internal/pipe/aur/aur.go:370 +0x14a
github.com/goreleaser/goreleaser/v2/internal/pipe/publish.Pipe.Run.Handle.func1(0xc000ba2060?)
	github.com/goreleaser/goreleaser/v2@v2.8.2/internal/middleware/errhandler/error.go:16 +0x1b
github.com/goreleaser/goreleaser/v2/internal/pipe/publish.Pipe.Run.PadLog.func2(0xc000844008)
	github.com/goreleaser/goreleaser/v2@v2.8.2/internal/middleware/logging/logging.go:47 +0x13d
github.com/goreleaser/goreleaser/v2/internal/middleware/skip.Maybe.func1(0xc000844008)
	github.com/goreleaser/goreleaser/v2@v2.8.2/internal/middleware/skip/skip.go:42 +0x10c
github.com/goreleaser/goreleaser/v2/internal/pipe/publish.Pipe.Run({{0xc0009c6b40?, 0xc002f770b0?, 0x5221160?}}, 0xc000844008)
	github.com/goreleaser/goreleaser/v2@v2.8.2/internal/pipe/publish/publish.go:86 +0x1a5
github.com/goreleaser/goreleaser/v2/cmd.releaseProject.func1.Handle.1(0xc00075d038?)
	github.com/goreleaser/goreleaser/v2@v2.8.2/internal/middleware/errhandler/error.go:16 +0x1b
github.com/goreleaser/goreleaser/v2/cmd.releaseProject.func1.Log.2(0xc000844008)
	github.com/goreleaser/goreleaser/v2@v2.8.2/internal/middleware/logging/logging.go:31 +0x132
github.com/goreleaser/goreleaser/v2/internal/middleware/skip.Maybe.func1(0xc000844008)
	github.com/goreleaser/goreleaser/v2@v2.8.2/internal/middleware/skip/skip.go:42 +0x10c
github.com/goreleaser/goreleaser/v2/cmd.releaseProject.func1()
	github.com/goreleaser/goreleaser/v2@v2.8.2/cmd/release.go:123 +0x182
github.com/caarlos0/ctrlc.(*Ctrlc).Run.func1()
	github.com/caarlos0/ctrlc@v1.2.0/ctrlc.go:48 +0x23
created by github.com/caarlos0/ctrlc.(*Ctrlc).Run in goroutine 1
	github.com/caarlos0/ctrlc@v1.2.0/ctrlc.go:47 +0x94
Error: The process '/opt/hostedtoolcache/goreleaser-action/2.8.2/x64/goreleaser' failed with exit code 2

```

- https://github.com/golangci/golangci-lint/blob/v2.1.0/.goreleaser.yml
- https://github.com/golangci/golangci-lint/actions/runs/14421579994/job/40444819022
- </div>

--- a/internal/artifact/artifact.go
+++ b/internal/artifact/artifact.go
@@ -77,6 +77,10 @@ const (
 	PkgBuild
 	// SrcInfo is an Arch Linux AUR .SRCINFO file.
 	SrcInfo
+	// SourcePkgBuild is an Arch Linux AUR PKGBUILD file for a source build.
+	SourcePkgBuild
+	// SourceSrcInfo is an Arch Linux AUR .SRCINFO file for a source build.
+	SourceSrcInfo
 	// KrewPluginManifest is a krew plugin manifest file.
 	KrewPluginManifest
 	// ScoopManifest is an uploadable scoop manifest file.
@@ -97,7 +101,10 @@ const (
 
 func (t Type) isUploadable() bool {
 	switch t {
-	case Binary, Metadata, SrcInfo, UniversalBinary:
+	case UniversalBinary, Binary, // See: [UploadableBinary].
+		Metadata,               // Local only.
+		SrcInfo, SourceSrcInfo, // It's always named `.SRCINFO`
+		PkgBuild, SourcePkgBuild: // It's always named `.PKGBUILD`
 		return false
 	default:
 		return true
@@ -138,9 +145,9 @@ func (t Type) String() string {
 		return "Scoop Manifest"
 	case SBOM:
 		return "SBOM"
-	case PkgBuild:
+	case PkgBuild, SourcePkgBuild:
 		return "PKGBUILD"
-	case SrcInfo:
+	case SrcInfo, SourceSrcInfo:
 		return "SRCINFO"
 	case PublishableChocolatey:
 		return "Chocolatey"

--- a/internal/artifact/artifact_test.go
+++ b/internal/artifact/artifact_test.go
@@ -1009,7 +1009,7 @@ func TestArtifactStringer(t *testing.T) {
 }
 
 func TestArtifactTypeStringer(t *testing.T) {
-	for i := 1; i <= 30; i++ {
+	for i := 1; i <= 32; i++ {
 		t.Run(fmt.Sprintf("type-%d-%s", i, Type(i).String()), func(t *testing.T) {
 			require.NotEqual(t, "unknown", Type(i).String())
 		})
@@ -1025,9 +1025,12 @@ func TestArtifactTypeIsUploadable(t *testing.T) {
 		int(Binary),
 		int(Metadata),
 		int(SrcInfo),
+		int(SourceSrcInfo),
+		int(PkgBuild),
+		int(SourcePkgBuild),
 		int(UniversalBinary),
 	}
-	for i := 1; i <= 30; i++ {
+	for i := 1; i <= 32; i++ {
 		up := Type(i).isUploadable()
 		t.Run(fmt.Sprintf("%s-%v", Type(i).String(), up), func(t *testing.T) {
 			if slices.Contains(nonUploadable, i) {

--- a/internal/pipe/aursources/aursources.go
+++ b/internal/pipe/aursources/aursources.go
@@ -145,13 +145,13 @@ func doRun(ctx *context.Context, aur config.AURSource, cl client.ReleaseURLTempl
 			name: "PKGBUILD",
 			tpl:  aurTemplateData,
 			ext:  ".pkgbuild",
-			kind: artifact.PkgBuild,
+			kind: artifact.SourcePkgBuild,
 		},
 		{
 			name: ".SRCINFO",
 			tpl:  srcInfoTemplate,
 			ext:  ".srcinfo",
-			kind: artifact.SrcInfo,
+			kind: artifact.SourceSrcInfo,
 		},
 	} {
 		pkgContent, err := buildPkgFile(ctx, aur, cl, archives, info.tpl)
@@ -323,8 +323,8 @@ func (Pipe) Publish(ctx *context.Context) error {
 	skips := pipe.SkipMemento{}
 	for _, pkgs := range ctx.Artifacts.Filter(
 		artifact.Or(
-			artifact.ByType(artifact.PkgBuild),
-			artifact.ByType(artifact.SrcInfo),
+			artifact.ByType(artifact.SourcePkgBuild),
+			artifact.ByType(artifact.SourceSrcInfo),
 		),
 	).GroupByID() {
 		err := doPublish(ctx, pkgs)

